trin-core/src/types/validation.rs

```diff
@@ -0,0 +1,93 @@
+use anyhow::anyhow;
+use async_trait::async_trait;
+use serde_json::{json, Value};
+
+use crate::jsonrpc::service::dispatch_infura_request;
+use crate::jsonrpc::types::{HistoryJsonRpcRequest, JsonRequest, Params};
+use crate::portalnet::types::content_key::IdentityContentKey;
+use crate::portalnet::types::messages::ByteList;
+
+// aka. HeaderOracle
+// This struct is responsible for dispatching cross-overlay-network requests
+// for data to perform validation. Currently, it just proxies these requests
+// on to infura.
+#[derive(Clone)]
+pub struct ValidationOracle {
+    pub infura_url: String,
+    // We could simply store the main portal jsonrpc tx channel here, rather than each
+    // individual channel. But my sense is that this will be more useful in terms of
+    // determining which subnetworks are actually available.
+    pub history_jsonrpc_tx: Option<tokio::sync::mpsc::UnboundedSender<HistoryJsonRpcRequest>>,
+    pub header_gossip_jsonrpc_tx: Option<bool>,
+    pub block_indices_jsonrpc_tx: Option<bool>,
+    pub state_jsonrpc_tx: Option<bool>,
+}
+
+impl Default for ValidationOracle {
+    fn default() -> Self {
+        Self {
+            infura_url: "https://mainnet.infura.io::443/v3/".to_string(),
```

Comment:
This extra colon looks wrong
```suggestion
            infura_url: "https://mainnet.infura.io:443/v3/".to_string(),
```